src/briefcase/commands/convert.py

```diff
@@ -0,0 +1,761 @@
+                test_entry_script,
+                test_path / f"{module_name}.py",
+            )
+        else:
+            copytree(project_dir / "tests", test_path)
+
+    def convert_app(
+        self,
+        tmp_path: Path,
+        template: str | None = None,
+        template_branch: str | None = None,
+        project_overrides: dict[str, str] | None = None,
+        **options,
+    ) -> None:
+        """Run the wizard in a temporary directory and copy the necessary files into the
+        project.
+
+        :param tmp_path: Temporary path that should contain the files generated by
+            cookiecutter.
+        :param template: The cookiecutter template to use.
+        :param template_branch: The git branch that the template should use.
+        """
+        version, template, branch = self.get_version_and_template_info(
+            template, template_branch
+        )
+        self.input.prompt()
+        self.input.prompt("Let's setup an existing project as a Briefcase app!")
+        context = self.build_context(template, branch, version, project_overrides)
+        self.prompt_divider()  # close the prompting section of output
+
+        self.warn_unused_overrides(project_overrides)
+
+        self.logger.info()
+        self.logger.info(
+            f"Generating required files to set up {context['formal_name']!r} with Briefcase",
+            prefix=context["app_name"],
+        )
+
+        # Create the project files
+        self.generate_template(
+            template=template,
+            branch=branch,
+            output_path=tmp_path,
+            extra_context=context,
+            allow_fallback=version.dev is not None,
+        )
+
+        app_path = context["app_name"]
+        self.logger.info(
+            f"Application {context['formal_name']!r} has been generated. To run your application, type:\n"
+            "\n"
+            f"cd {app_path}\n"
+            "briefcase dev"
+        )
+
+        project_dir = tmp_path / context["app_name"]
+        self.migrate_necessary_files(
+            project_dir, context["test_source_dir"], context["module_name"]
+        )
+
+    def validate_pyproject_file(self) -> None:
+        """Cannot setup new app if it already has briefcase settings in pyproject."""
+        if (self.base_path / "pyproject.toml").exists():
+            with open(self.base_path / "pyproject.toml", "rb") as file:
+                pyproject = tomllib.load(file)
+
+            if "tool" in pyproject and "briefcase" in pyproject["tool"]:
+                raise BriefcaseCommandError(
+                    f"[tool.briefcase] already in {self.base_path / 'pyproject.toml'}."
+                    " Cannot initialise briefcase for an application that already is packaged with briefcase."
+                )
+
+    def validate_not_empty_project(self) -> None:
+        """Cannot setup new app for a project containing only a child directory with log
+        files or no child directories."""
```

Comment:
I'm not sure that this validation step is required. This validation will pass if there are *any* subdirectories that don't contain logs - so a `venv` or `tmp` folder would be enough to make it pass.

In addition, one of the first things validated in `build_app_context` is the source folder, which will also validate that a directory exists, but will be a more rigorous check, as it validates that a *specific* directory exists. 

The only case I can see that this *will* catch is a truly "bare" project that only contains `pyproject.toml` and `myapp.py` (and maybe a `README` or other stray files) - but that project will get flagged as soon as you try to find the source folder.

On balance, it seems to me that while this validation step isn't technically *incorrect*, it isn't actually adding any useful validation in practice.

Is there some other use case that is a source of concern that this validating?

Reply:
We added this because there was this one time when we tried running the wizard in an empty directory twice. The first time, we got an error in the validation step, while we did not the second time because of the log-directory, which we thought were a bit confusing.